app/src/main/java/com/mealsmadeeasy/ui/splash/SplashActivity.kt

```diff
@@ -1,15 +1,122 @@
+
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
+import android.support.design.widget.Snackbar
+import android.widget.Button
+import com.google.android.gms.common.SignInButton
+import kotlinx.android.synthetic.main.activity_splash.*
+
+
+class SplashActivity : BaseActivity(), View.OnClickListener {
+
+    private val TAG = "SplashActivity"
+    private val RC_SIGN_IN = 9001
+
+    private var mAuth: FirebaseAuth? = null
+    private var mGoogleSignInClient: GoogleSignInClient? = null
 
-class SplashActivity : BaseActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        startActivity(HomeActivity.newIntent(this))
-        finish()
+        setContentView(R.layout.activity_splash)
+        findViewById<SignInButton>(R.id.sign_in_button).setOnClickListener(this)
+
+        // Configure sign-in to request the user's ID, email address, and basic
+        // profile. ID and basic profile are included in DEFAULT_SIGN_IN.
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id)) //OAuth 2.0 client ID
+                .requestEmail()
+                .build()
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        mGoogleSignInClient = GoogleSignIn.getClient(this, gso)
+        mAuth = FirebaseAuth.getInstance()
     }
 
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+        val currentUser = mAuth?.getCurrentUser()
+        updateUI(currentUser)
+    }
+
+    private fun signIn() {
+        val signInIntent = mGoogleSignInClient?.getSignInIntent()
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)
+                firebaseAuthWithGoogle(account)
+            } catch (e: ApiException) {
+                // Google Sign In failed, update UI appropriately
+                Log.w(TAG, "Google sign in failed", e)
+                updateUI(null)
+            }
+
+        }
+    }
+
+    private fun firebaseAuthWithGoogle(acct: GoogleSignInAccount) {
+        Log.d(TAG, "firebaseAuthWithGoogle:" + acct.id!!)
```

Comment:
Is there a reason we need to keep the debug logs here? It feels like it's not adding anything useful and could leak potentially sensitive information to logcat which isn't great.